### DISCUSSION TITLE
Live z Azure Bicep

### DISCUSSION THREAD

**Discussion by kaluzaaa** [AzureCommunityPL/group#105] [Community Content]
Jutro (09.10) od 20:30 będę rozpoznawał w boju i na żywo uczył się nowego języka (DSL) do deployemntu zasobów na Azure, czyli https://github.com/Azure/bicep.

W teorii jest to nakładka na ARM, która ma być o niebo przyjemniejsza. Jak w praktyce to się przekonamy na żywo.

Można oglądać na [YT](https://www.youtube.com/watch?v=WsLIUXiTCy4&ab_channel=Patoarchitekci) lub [FB](https://www.facebook.com/kaluznypro/posts/4621462457896401)

**Comment by piotrgwiazda** [comment]
Czyli to konwersja z jakiegoś formatu nie-JSON (będzie miał komentarze!) Do
JSONa? I tyle, ze wszystkimi wadami ARMa?
W sumie to można sobie użyć HCL czy nawet YAML, po co kolejny język? Na oko
różni się o tyle od HCL, żeby osoba znająca terraforma bez przerwy się
myliła :-)

czw., 8 paź 2020, 18:55 użytkownik Łukasz Kałużny <notifications@github.com>
napisał:

**Reply by kaluzaaa** [reply]
Ten json to największy wrzód na arm ;-) idea jest taka, żeby to było przyjemniejsze niż to co teraz ma w json (szczególnie wszystkie funkcje). Powiem tak, dobrze, że 4 z 5 poprzednich pomysłów jak to ma wyglądać nie ujrzało światła dziennego ;-)

**Reply by ptrstpp950** [reply]
Ja się muszę zabrać kiedyś za Pulumi. Twardy dobry kod :)

**Reply by kaluzaaa** [reply]
@ptrstpp950 ja ciągle mam mieszane uczucia co do pulumi ;-) Szczególnie, że raz ktoś napisze korzystając z ich zabawek w jednym języku, potem w innym i co ten biedny devops inżnier, który ma robić, a zwykle z kodowaniem jest potem jak ze świnką morską :-D

**Reply by kamil-mrzyglod** [reply]
@piotrgwiazda W sensie o jakich wadach ARM(ARM Template) mówisz? JSON jest rozwleczony ale czy gorszy od rozwleczonego YAML to już subiektywne różnice 😉 Mnie strasznie wkurzały Nested Templates ale odkąd na horyzoncie są _template specs_ to chyba da się z tym żyć. Na komentarze też sobie sposób wymyślili - https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-best-practices#resources. Z drugiej strony jak się używa PowerShell do deploymentu to tam chyba JSON.NET albo pokrewna biblioteka radzi sobie z komentarzami więc można radośnie sobie traktować szablon jako JSONC.

EDIT: No nie, _template specs_ bardziej rozwiązują problem, który Terraform rozwiązywał czyli modułowość ;)
